File: 프로그래머스/lv1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.py
def solution(dartResult):
    scoreList=[]
    score = ''
    for i in dartResult:
        if i == '#':
            scoreList[-1]*=-1
            continue
        if i == '*':
            if len(scoreList) > 1:
                scoreList[-1]*=2
                scoreList[-2]*=2
            else:
                scoreList[-1]*=2

            continue
        if i.isdigit():
            score+=i
        else:
            intScore = int(score)
            if i == 'D':
                intScore **= 2
            if i == 'T':
                intScore **= 3
            scoreList.append(intScore)
            score=''
    return sum(scoreList)

File: 프로그래머스/lv1/17682. ［1차］ 다트 게임/［1차］ 다트 게임.js
function solution(dartResult) {
    let number = '';
    let answer = [];
    
    for(let i=0; i < dartResult.length; i++){
        if(isNaN(dartResult[i]) === false) number+=dartResult[i];
        else{
            if(dartResult[i] === "S"){
                answer.push(Number(number));
                number = '';
            } else if(dartResult[i] === "D"){
                answer.push(Number(number)**2);
                number = '';
            } else if(dartResult[i] === "T"){
                answer.push(Number(number)**3);
                number = '';
            } else if(dartResult[i] === "*"){
                answer[answer.length-1] = answer.at(-1)*2;
                answer[answer.length-2] = answer.at(-2)*2;
            } else if(dartResult[i] === "#"){
                answer[answer.length-1] = -answer.at(-1);
            };
        };
    };
    return answer.reduce((a,b) => a+b);
}